### DISCUSSION TITLE
what's the difference between LONG_COMPAT and LEGACY machine modes ?

### DISCUSSION THREAD

**Discussion by 440bx** [zyantific/zydis#495] [Q&A] [answered]
Hello, 

the ZydisMachineMode enumeration includes a ZYDIS_MACHINE_MODE_LONG_COMPAT_32, ZYDIS_MACHINE_MODE_LEGACY_32 and a "parallel" pair ZYDIS_MACHINE_MODE_LONG_COMPAT_16, ZYDIS_MACHINE_MODE_LEGACY_16.

What's the difference between the COMPAT vs LEGACY mode ?  

For disassembling Windows and Linux programs, which one should be used ? and why ?

Thank you for your help.

**Comment by flobernd** [comment, marked as answer]
https://en.m.wikipedia.org/wiki/X86

Check: Operating Modes.

I would recommend LONG_64 or LONG_COMPAT_32 for recently modern binaries.

**Comment by 440bx** [comment]
Thank you Florian.
